lib/media/track/remotevideotrack.js

```diff
@@ -47,23 +48,41 @@ class RemoteVideoTrack extends RemoteMediaVideoTrack {
       _documentVisibilityTurnOffCleanup: {
         value: null,
         writable: true
-      }
+      },
+      _intersectionObserver: {
+        value: options.IntersectionObserver ? new options.IntersectionObserver(entries => {
+          let shouldSetRenderHint = false;
+          entries.forEach(entry => {
+            if (entry.target.isIntersecting !== entry.isIntersecting) {
+              entry.target.isIntersecting = entry.isIntersecting;
+              shouldSetRenderHint = true;
+            }
+          });
+          if (shouldSetRenderHint) {
+            updateRenderHints(this);
+          }
+        }, { threshold: 0.25 }) : null
+      },
     });
   }
 
   attach(el) {
     const result = super.attach(el);
+    this._intersectionObserver.observe(result);
     if (this._enableDocumentVisibilityTurnOff) {
       this._documentVisibilityTurnOffCleanup = this._documentVisibilityTurnOffCleanup
         || setupDocumentVisibilityTurnOff(this);
     }
-
     return result;
   }
 
   detach(el) {
     const result = super.detach(el);
-
+    if (Array.isArray(result)) {
+      result.forEach(element => this._intersectionObserver.unobserve(element));
```

Comment:
now that `this._intersectionObserver` can be null we have to check if its not null